sound/soc/sof/pcm.c

```diff
@@ -213,7 +247,14 @@ static int sof_pcm_hw_params(struct snd_soc_component *component,
 
 	dev_dbg(component->dev, "stream_tag %d", pcm.params.stream_tag);
 
-	/* send IPC to the DSP */
+	/* if this is a repeated hw_params without hw_free, skip setting up widgets */
+	if (!spcm->stream[substream->stream].list) {
```

Comment:
Do we have such a test case for repeated hw_params ? we need add it to shop list if not

Reply:
yes, see the comment on OSS emulation on line 175.